### PR TITLE
Add datasets required for tutorials

### PR DESCRIPTION
**Pull request by lee-suddaby**
Adding the necessary datasets for the tutorials due to be added to the package.

### CODE REVIEW

**Comment by lee-suddaby** [comment]
@mine-cetinkaya-rundel

File: R/accidents.R [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#' Road Safety Data - Accidents 2018
+#'
+#' Road accidents and their causes are of particular importance to road safety experts
+#' looking to prevent them.
+#'
+#' The data come from the UK Government at
+#' [data.gov.uk](https://data.gov.uk/dataset/cb7ae6f0-4be6-4935-9277-47e5ce24a11f/road-safety-data/datafile/36f1658e-b709-47e7-9f56-cca7aefeb8fe/preview)
+#' It's been modified to better serve the goals of introductory data science education.
+#'
+#' We can use it to explore
```
Comment (mine-cetinkaya-rundel) [inline R/accidents.R]: I think we can remove this sentence fragment.

File: R/dcbikeshare.R [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#' Bike Rentals In D.C.
+#'
+#' Bike sharing systems are new generation of traditional bike rentals where whole
+#' process from membership, rental and return back has become automatic. Through these
+#' systems, user is able to easily rent a bike from a particular position and return
+#' back at another position. Currently, there are about over 500 bike-sharing programs
```
Comment (mine-cetinkaya-rundel) [inline R/dcbikeshare.R]: What does "Currently" refer to here. Should we say "As of 2020"?

File: R/dcbikeshare.R [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+#' Apart from interesting real world applications of bike sharing systems, the
+#' of data being generated by these systems make them attractive for the research.
+#' Opposed to other transport services such as bus or subway, the duration of travel,
+#' departure and arrival position is explicitly recorded in these systems. This
+#' feature turns bike sharing system into a virtual sensor network that can be used
+#' for sensing mobility in the city. Hence, it is expected that most of important
+#' events in the city could be detected via monitoring these data.
+#'
+#' The dataset comes from
+#' [UC Irvine](http://archive.ics.uci.edu/ml/datasets/Bike+Sharing+Dataset). Sources of the
```
Comment (mine-cetinkaya-rundel) [inline R/dcbikeshare.R]:
```suggestion
#' [UCI Machine Learning Repository](http://archive.ics.uci.edu/ml/datasets/Bike+Sharing+Dataset). Sources of the
```

File: R/dcbikeshare.R [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+#' data itself are
+#' [Capital Bikeshare](https://www.capitalbikeshare.com/system-data),
+#'  [freemeteo.co.uk](http://www.freemeteo.co.uk) for weather data, and
+#'  [dv.gov](http://dchr.dc.gov/page/holiday-schedule) for the holiday schedule.
```
Comment (mine-cetinkaya-rundel) [inline R/dcbikeshare.R]: Remove empty spaces at the beginning of lines to align with line 22

File: R/dcbikeshare.R [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+#'
+#' The data has been modified to better serve the goals of introductory data science education.
```
Comment (mine-cetinkaya-rundel) [inline R/dcbikeshare.R]:
```suggestion
#' The data have been modified to better serve the goals of introductory data science education.
```

File: R/gss.R [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+#'
+#' @source https://gss.norc.org/Get-The-Data
+#' @format A tibble with 2,867 rows and 9 variables:
+#' \describe{
+#'   \item{harass5}{"Over the past five years, have you been harassed by your superiors or co-workers at your job, for example, have you experienced any bullying, physical or psychological abuse?"}
+#'   \item{emailmin}{Number of minutes spent on email weekly, extra to the hours in emailhrs (e.g. emailmin = 30 for 2.5 hours on email).}
+#'   \item{emailhr}{Number of hours spent on email weekly.}
+#'   \item{educ}{Number of years in education.}
+#'   \item{polviews}{Political views. Possible answers are Extremely liberal, Liberal, Slightly liberal, Moderate, Slghtly conservative, Conservative, Extrmly conservative.}
+#'   \item{advfront}{"Even if it brings no immediate benefits, scientific research that advances the frontiers of knowledge is necessary and should be supported by the federal government."}
```
Comment (mine-cetinkaya-rundel) [inline R/gss.R]:
```suggestion
#'   \item{advfront}{Response to the question "Even if it brings no immediate benefits, scientific research that advances the frontiers of knowledge is necessary and should be supported by the federal government."}
```

Comment (mine-cetinkaya-rundel) [reply R/gss.R]: Also say what the possible answers are, are they yes/no, or something else?

File: R/gss.R [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+#'
+#' The data come from the [NORC at the University of Chicago](https://gss.norc.org/Get-The-Data)
+#'
+#' The data has been modified to better serve the goals of introductory data science education.
+#' Only 9 of the original 935 variables have been selected for this dataset.
+#'
+#' @source https://gss.norc.org/Get-The-Data
+#' @format A tibble with 2,867 rows and 9 variables:
+#' \describe{
+#'   \item{harass5}{"Over the past five years, have you been harassed by your superiors or co-workers at your job, for example, have you experienced any bullying, physical or psychological abuse?"}
```
Comment (mine-cetinkaya-rundel) [inline R/gss.R]:
```suggestion
#'   \item{harass5}{Response to the question "Over the past five years, have you been harassed by your superiors or co-workers at your job, for example, have you experienced any bullying, physical or psychological abuse?"}
```

Comment (mine-cetinkaya-rundel) [reply R/gss.R]: Also say what the possible answers are, are they yes/no, or something else?

File: R/gss.R [hunk 8]
```diff
@@ -0,0 +1,38 @@
+#' GSS - General Social Survey
```
Comment (mine-cetinkaya-rundel) [inline R/gss.R]: Did we mention in the homework/tutorials which year these data are from? If so, we should add it here and make the title "Data from the XXXX General Social Survey (GSS)". If this is not mentioned somewhere let me know and I'll dig up my records and find it.

File: R/lego_sales.R [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+#'   \item{year}{Year of purchase}
+#'   \item{name}{ Name of lego set purchased}
+#'   \item{pieces}{Number of pieces of legos in set purchased}
+#'   \item{us_price}{Price of set purchase in US Dollars}
+#'   \item{image_url}{Image URL of lego set purchased}
+#'   \item{quantity}{Quantity of lego set(s) purchased}
+#' }
+#' @examples
+#'
+#' library(tidyverse)
```
Comment (mine-cetinkaya-rundel) [inline R/lego_sales.R]: Let's make this `library(dplyr)` so we're not depending on the entire tidyverse package for the package

File: DESCRIPTION [hunk 10]
```diff
@@ -13,7 +13,8 @@ Encoding: UTF-8
 LazyData: true
 Suggests:
     ggplot2,
-    learnr
+    learnr,
+    tidyverse
```
Comment (mine-cetinkaya-rundel) [inline DESCRIPTION:18]: Use `dplyr` instead of `tidyverse` here.

File: R/pac_2020.R [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+#' Every election cycle brings its own brand of excitement – and lots of money.
+#' Political donations are of particular interest to political scientists and
+#' other researchers studying politics and voting patterns.
+#' They are also of interest to citizens who want to stay informed of how much
+#' money their candidates raise and where that money comes from.
+#'
+#' In the United States, “only American citizens (and immigrants with green
+#' cards) can contribute to federal politics, but the American divisions of
+#' foreign companies can form political action committees (PACs) and collect
+#' contributions from their American employees.” (source: OpenSecrets.org)
```
Comment (mine-cetinkaya-rundel) [inline R/pac_2020.R]:
```suggestion
#' contributions from their American employees.” (Source: [OpenSecrets.org](https://www.opensecrets.org/political-action-committees-pacs/foreign-connected-pacs))
```

File: R/pac_2020.R [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+#'
+#' The data come from the
+#' [OpenSecrets.org](https://www.opensecrets.org/political-action-committees-pacs/foreign-connected-pacs).
+#'
+#' The data has been scraped from the OpenSecrets.org website.
```
Comment (mine-cetinkaya-rundel) [inline R/pac_2020.R]: Add date when these data were scraped, since they change periodically

File: R/pac_2020.R [hunk 13]
```diff
@@ -0,0 +1,40 @@
+#' Political Action Committee Contributions
```
Comment (mine-cetinkaya-rundel) [inline R/pac_2020.R]:
```suggestion
#' Foreign-Connected PAC Contributions
```

File: R/pac_2020_raw.R [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+#'   \item{Country of Origin/Parent Company}{Country of origin and (foreign) parent company}
+#'   \item{Total}{Total PAC contributions to all parties}
+#'   \item{Dems}{PAC contributions to the Democratic Party}
+#'   \item{Repubs}{PAC contributions to the Republican Party}
+#' }
+#' @examples
+#'
+#' library(tidyverse)
+#'
+#' glimpse(pac_2020_raw)
```
Comment (mine-cetinkaya-rundel) [inline R/pac_2020_raw.R]:
```suggestion
#' pac_2020_raw
```

Comment (mine-cetinkaya-rundel) [reply R/pac_2020_raw.R]: Since it's not too many columns anyway...

File: R/lego_sales.R [hunk 15]
```diff
@@ -0,0 +1,32 @@
+#' Sales of Lego sets
```
Comment (mine-cetinkaya-rundel) [inline R/lego_sales.R]:
```suggestion
#' Sales of Lego sets (simulated)
```

File: R/lego_sales.R [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+#'
+#' Here we have data from Lego sales in 2018 for a sample of customers who bought
+#' Legos in the United States.
```
Comment (mine-cetinkaya-rundel) [inline R/lego_sales.R]:
```suggestion
#' Simulated data based on real prices for Lego sales in 2018 in the United States. 
#' Names and hobbies have been randomly generated
```

**Review by mine-cetinkaya-rundel — CHANGES_REQUESTED** [review]
Looking good!

Some minor edits, and then one bigger one about the foreign connected PAC datasets -- there is a way to combine documentation for a group of connected functions/datasets using the @seealso tag in Roxygen. I also wonder if some of these datasets should just be constructed at the beginning of the tutorial in the setup chunk or if they're valuable to have as separate datasets here. Once you go through the other edits, I'll do this combining/sorting through and let you know my approach.

Also, I made some edits to master while you were working on this PR as well as made some direct suggestions here. As a workflow I would recommend

- Read through comments here and accept the suggestions
- Then, in RStudio run `pr_sync()` (see the bottom of [this vignette](https://usethis.r-lib.org/articles/articles/pr-functions.html) for more on that)
- Then make changes in your PR in RStudio

File: R/dcbikeshare.R (hunk 2 above)
Comment (lee-suddaby) [reply R/dcbikeshare.R]: I took that straight from the tutorial on introds.org, so honestly I don't know.

However, a quick Google search reveals it's a lot higher than that now, so I'll change this sentence and add a year for reference.

File: R/gss.R (hunk 8 above)
Comment (lee-suddaby) [reply R/gss.R]: It was the 2016 GSS.

File: DESCRIPTION (hunk 10 above)
Comment (lee-suddaby) [reply DESCRIPTION:18]: @mine-cetinkaya-rundel Might be better to leave this as is, since the tutorials all use the tidyverse package.

**Comment by lee-suddaby** [comment]
@mine-cetinkaya-rundel I've made those suggested changes, except for the one about removing the tidyverse dependency, which I've commented on above.

So I think all that's left to do here is to experiment with and decide on an approach for the foreign-connected PAC datasets.

File: DESCRIPTION (hunk 10 above)
Comment (mine-cetinkaya-rundel) [reply DESCRIPTION:18]: You might be right, but it seems like RStudio has a way of catching dependencies of a tutorial and prompting the user to install if needed. I'm going to try with and without in a fresh Cloud session to see what happens.

**Review by mine-cetinkaya-rundel — APPROVED** [review]

**Pull request merged**